src/server/lib/__tests__/headers.test.ts

```diff
@@ -0,0 +1,74 @@
+import { ValidRoutePathsArray } from '@/shared/model/Routes';
+import Redis from 'ioredis-mock';
+import request from 'supertest';
+import { getServerInstance } from './sharedConfig';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+
+// Override the default 5s max timeout for these tests because Supertest takes some time to run.
+jest.setTimeout(10000);
+
+describe('Content Security Policy headers', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetModules();
+    jest.mock('aws-sdk');
+    jest.mock('@/server/lib/getAssets', () => ({
+      getAssets: () => ({
+        main: { js: 'mocked' },
+        vendors: { js: 'mocked' },
+        runtime: { js: 'mocked' },
+      }),
+    }));
+
+    jest.mock('@/server/lib/serverSideLogger', () => ({
+      logger: {
+        info: jest.fn(),
+        error: jest.fn(),
+      },
+    }));
+
+    jest.mock('@/server/lib/redis/redisClient', () => new Redis());
+
+    jest.mock('@/server/lib/middleware/login', () => ({
+      loginMiddleware: jest.fn((req, res, next) => next()),
+    }));
+    jest.mock('@/server/lib/IDAPIFetch');
+  });
+
+  it('should return CSP headers for routes', async () => {
```

Comment:
could be worth explicitly saying this covers all routes in the test description here